src/web/app/components/course-edit-form/course-edit-form.component.spec.ts

```diff
@@ -215,4 +216,188 @@ describe('CourseEditFormComponent', () => {
     expect(mockModalRef.componentInstance.activeCourses[0]).toEqual(testCourse);
     expect(mockModalRef.componentInstance.courseToFeedbackSession[testCourse.courseId]).toEqual([testFeedbackSession]);
   });
+
+  it('should handle errors in copyCourseHandler when promise is rejected', () => {
+    component.formModel = DEFAULT_COURSE_ADD_FORM_MODEL();
+    component.formMode = CourseEditFormMode.ADD;
+    fixture.detectChanges();
+
+    const mockModalRef: any = createMockNgbModalRef({
+      isCopyFromOtherSession: false,
+      courses: [],
+      courseToFeedbackSession: {},
+      fetchFeedbackSessionsEvent: new EventEmitter<string>(),
+    }, Promise.reject(new Error(errorMssg)));
+    jest.spyOn(ngbModal, 'open').mockReturnValue(mockModalRef);
+
+    component.copyCourseHandler();
+
+    expect(spyStatusMessageService.showErrorToast()).toBe(errorMssg);
```

Comment:
this is always true as showErrorToast always returns errorMssg, we should instead be checking if the showErrorToast function has been called with errorMsg, using the `toHaveBeenCalledWith` function

Reply:
I was working on writing a test case for the 'copyCourseHandler' function, specifically for the scenario where the promise is rejected. While doing this, I noticed something that I think might be worth considering for a change.

Currently, in the promise rejection handling part of the 'copyCourseHandler,' we have defined the parameter type as 'ErrorMessageOutput' when dealing with the rejection. However, promise rejection only returns a string.

Based on lint rules should I change the 
'(resp: ErrorMessageOutput) => this.statusMessageService.showErrorToast(resp.error.message)' to 
'(resp: Error) => this.statusMessageService.showErrorToast(resp.message)'

lint rule for promise rejection: https://eslint.org/docs/latest/rules/prefer-promise-reject-errors

Here's the relevant code snippet for reference:

```typescript
modalRef.result.then(
    (result: CopyCourseModalResult) => this.copyCourseEvent.emit(result),
    (resp: ErrorMessageOutput) => this.statusMessageService.showErrorToast(resp.error.message)
);

Reply:
hey @lunoos, while the eslint suggestion is valid, currently all our backend error responses follow the `ErrorMessageOutput` type, hence I do not think we should change it as that is what the actual backend returns. We should mock it correctly such that it returns the appropriate type for example:

```
    const customError: ErrorMessageOutput = {
      error: {
        message: errorMssg,
      },
      status: 0
    };

    const mockModalRef: any = createMockNgbModalRef({
      isCopyFromOtherSession: false,
      courses: [],
      courseToFeedbackSession: {},
      fetchFeedbackSessionsEvent: new EventEmitter<string>(),
    }, Promise.reject(customError));
```

Reply:
yes, that's fine, we can add a comment to disable the lint rule for this particular line